### ISSUE TITLE
Jooby "Hello World" triggers backend error in IDE

### ISSUE THREAD

**Issue by fwgreen** [ceylon/ceylon-ide-eclipse#1869]
This compiles and runs from the command line, crashes inside Eclipse
```Ceylon
import org.jooby {
  ...
}

shared class App() extends Jooby() {
  
  get("/", () => "Hello world");
}

shared void run() {
  
  Jooby.run(() => App(), *process.arguments);
}
```
```
Description	Path	Resource	Location	Type
The Ceylon Java backend compilation failed
  with a throwable : java.lang.NullPointerException
	at com.redhat.ceylon.langtools.tools.javac.comp.Flow$FlowAnalyzer.visitApply(Flow.java:1250)
	at com.redhat.ceylon.langtools.tools.javac.tree.JCTree$JCMethodInvocation.accept(JCTree.java:1464)
	at com.redhat.ceylon.langtools.tools.javac.tree.TreeScanner.scan(TreeScanner.java:49)
	at com.redhat.ceylon.langtools.tools.javac.comp.Flow$BaseAnalyzer.scan(Flow.java:412)
	at com.redhat.ceylon.langtools.tools.javac.comp.Flow$FlowAnalyzer.visitReturn(Flow.java:1225)
	at com.redhat.ceylon.langtools.tools.javac.tree.JCTree$JCReturn.accept(JCTree.java:1383)
	at com.redhat.ceylon.langtools.tools.javac.tree.TreeScanner.scan(TreeScanner.java:49)
	at com.redhat.ceylon.langtools.tools.javac.comp.Flow$BaseAnalyzer.scan(Flow.java:412)
	at com.redhat.ceylon.langtools.tools.javac.tree.TreeScanner.scan(TreeScanner.java:57)
	at com.redhat.ceylon.langtools.tools.javac.comp.Flow$FlowAnalyzer.visitBlock(Flow.java:1010)
	at com.redhat.ceylon.langtools.tools.javac.tree.JCTree$JCBlock.accept(JCTree.java:908)
	at com.redhat.ceylon.langtools.tools.javac.tree.TreeScanner.scan(TreeScanner.java:49)
	at com.redhat.ceylon.langtools.tools.javac.comp.Flow$BaseAnalyzer.scan(Flow.java:412)
	at com.redhat.ceylon.langtools.tools.javac.comp.Flow$FlowAnalyzer.visitMethodDef(Flow.java:977)
	at com.redhat.ceylon.langtools.tools.javac.tree.JCTree$JCMethodDecl.accept(JCTree.java:777)
	at com.redhat.ceylon.langtools.tools.javac.tree.TreeScanner.scan(TreeScanner.java:49)
	at com.redhat.ceylon.langtools.tools.javac.comp.Flow$BaseAnalyzer.scan(Flow.java:412)
	at com.redhat.ceylon.langtools.tools.javac.comp.Flow$FlowAnalyzer.visitClassDef(Flow.java:940)
	at com.redhat.ceylon.langtools.tools.javac.tree.JCTree$JCClassDecl.accept(JCTree.java:692)
	at com.redhat.ceylon.langtools.tools.javac.tree.TreeScanner.scan(TreeScanner.java:49)
	at com.redhat.ceylon.langtools.tools.javac.comp.Flow$BaseAnalyzer.scan(Flow.java:412)
	at com.redhat.ceylon.langtools.tools.javac.comp.Flow$FlowAnalyzer.visitNewClass(Flow.java:1279)
	at com.redhat.ceylon.langtools.tools.javac.tree.JCTree$JCNewClass.accept(JCTree.java:1515)
	at com.redhat.ceylon.langtools.tools.javac.tree.TreeScanner.scan(TreeScanner.java:49)
	at com.redhat.ceylon.langtools.tools.javac.comp.Flow$BaseAnalyzer.scan(Flow.java:412)
	at com.redhat.ceylon.langtools.tools.javac.tree.TreeScanner.scan(TreeScanner.java:57)
	at com.redhat.ceylon.langtools.tools.javac.comp.Flow$FlowAnalyzer.visitApply(Flow.java:1249)
	at com.redhat.ceylon.langtools.tools.javac.tree.JCTree$JCMethodInvocation.accept(JCTree.java:1464)
	at com.redhat.ceylon.langtools.tools.javac.tree.TreeScanner.scan(TreeScanner.java:49)
	at com.redhat.ceylon.langtools.tools.javac.comp.Flow$BaseAnalyzer.scan(Flow.java:412)
	at com.redhat.ceylon.langtools.tools.javac.tree.TreeScanner.visitExec(TreeScanner.java:175)
	at com.redhat.ceylon.langtools.tools.javac.tree.JCTree$JCExpressionStatement.accept(JCTree.java:1295)
	at com.redhat.ceylon.langtools.tools.javac.tree.TreeScanner.scan(TreeScanner.java:49)
	at com.redhat.ceylon.langtools.tools.javac.comp.Flow$BaseAnalyzer.scan(Flow.java:412)
	at com.redhat.ceylon.langtools.tools.javac.tree.TreeScanner.scan(TreeScanner.java:57)
	at com.redhat.ceylon.langtools.tools.javac.comp.Flow$FlowAnalyzer.visitBlock(Flow.java:1010)
	at com.redhat.ceylon.langtools.tools.javac.tree.JCTree$JCBlock.accept(JCTree.java:908)
	at com.redhat.ceylon.langtools.tools.javac.tree.TreeScanner.scan(TreeScanner.java:49)
	at com.redhat.ceylon.langtools.tools.javac.comp.Flow$BaseAnalyzer.scan(Flow.java:412)
	at com.redhat.ceylon.langtools.tools.javac.comp.Flow$FlowAnalyzer.visitMethodDef(Flow.java:977)
	at com.redhat.ceylon.langtools.tools.javac.tree.JCTree$JCMethodDecl.accept(JCTree.java:777)
	at com.redhat.ceylon.langtools.tools.javac.tree.TreeScanner.scan(TreeScanner.java:49)
	at com.redhat.ceylon.langtools.tools.javac.comp.Flow$BaseAnalyzer.scan(Flow.java:412)
	at com.redhat.ceylon.langtools.tools.javac.comp.Flow$FlowAnalyzer.visitClassDef(Flow.java:940)
	at com.redhat.ceylon.langtools.tools.javac.tree.JCTree$JCClassDecl.accept(JCTree.java:692)
	at com.redhat.ceylon.langtools.tools.javac.tree.TreeScanner.scan(TreeScanner.java:49)
	at com.redhat.ceylon.langtools.tools.javac.comp.Flow$BaseAnalyzer.scan(Flow.java:412)
	at com.redhat.ceylon.langtools.tools.javac.comp.Flow$FlowAnalyzer.analyzeTree(Flow.java:1342)
	at com.redhat.ceylon.langtools.tools.javac.comp.Flow$FlowAnalyzer.analyzeTree(Flow.java:1332)
	at com.redhat.ceylon.langtools.tools.javac.comp.Flow.analyzeTree(Flow.java:219)
	at com.redhat.ceylon.langtools.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1330)
	at com.redhat.ceylon.compiler.java.tools.LanguageCompiler.flow(LanguageCompiler.java:862)
	at com.redhat.ceylon.langtools.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1299)
	at com.redhat.ceylon.langtools.tools.javac.main.JavaCompiler.compile2(JavaCompiler.java:904)
	at com.redhat.ceylon.langtools.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:862)
	at com.redhat.ceylon.compiler.java.tools.LanguageCompiler.compile(LanguageCompiler.java:272)
	at com.redhat.ceylon.compiler.java.launcher.Main.compile(Main.java:659)
	at com.redhat.ceylon.langtools.tools.javac.api.JavacTaskImpl.doCall(JavacTaskImpl.java:132)
	at com.redhat.ceylon.langtools.tools.javac.api.JavacTaskImpl.call(JavacTaskImpl.java:141)
	at com.redhat.ceylon.compiler.java.tools.CeyloncTaskImpl.call(CeyloncTaskImpl.java:59)
	at com.redhat.ceylon.eclipse.core.builder.CeylonBuilder.compile(CeylonBuilder.java:2663)
	at com.redhat.ceylon.eclipse.core.builder.CeylonBuilder.generateBinaries(CeylonBuilder.java:2392)
	at com.redhat.ceylon.eclipse.core.builder.CeylonBuilder.access$100(CeylonBuilder.java:210)
	at com.redhat.ceylon.eclipse.core.builder.CeylonBuilder$4.call(CeylonBuilder.java:1075)
	at com.redhat.ceylon.eclipse.core.builder.CeylonBuilder$4.call(CeylonBuilder.java:1072)
	at com.redhat.ceylon.eclipse.core.builder.CeylonBuilder.doWithCeylonModelCaching(CeylonBuilder.java:260)
	at com.redhat.ceylon.eclipse.core.builder.CeylonBuilder.build(CeylonBuilder.java:1072)
	at org.eclipse.core.internal.events.BuildManager$2.run(BuildManager.java:734)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:205)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:245)
	at org.eclipse.core.internal.events.BuildManager$1.run(BuildManager.java:300)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:303)
	at org.eclipse.core.internal.events.BuildManager.basicBuildLoop(BuildManager.java:359)
	at org.eclipse.core.internal.events.BuildManager.build(BuildManager.java:382)
	at org.eclipse.core.internal.events.AutoBuildJob.doBuild(AutoBuildJob.java:144)
	at org.eclipse.core.internal.events.AutoBuildJob.run(AutoBuildJob.java:235)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:55)
		Cooby	line 0	Ceylon Backend Error
```
Maybe this is related to Lambdas as using an alternative API compiles...
```Ceylon
shared class App() extends Jooby() {
  
  get("/", (req, res) => res.send("Hello world"));
}
```

**Comment by gavinking** [comment]
@fwgreen could you please show us your module descriptor? Thanks.

**Comment by fwgreen** [comment]
Sorry about that.
```Ceylon
native("jvm")
module cooby "1.0.0" {
  shared import maven:"org.jooby:jooby-undertow" "1.1.0";
}
```

**Comment by gavinking** [comment]
@fwgreen I think you must also have an overrides file, is that right?

**Comment by gavinking** [comment]
Oh no, I see, you're using "fully export maven dependencies", OK, now I can reproduce.

**Comment by gavinking** [comment]
So yeah, it compiles and runs just fine in IntelliJ, but crashes the compiler in Eclipse :-(

**Comment by gavinking** [comment]
As noted by @fwgreen, it works with this:

```ceylon
shared class App() extends Jooby() {
  get("/", (req, res) => res.send("Hello world"));
}
```

So it's definitely something to do with lambda conversions in Eclipse.

**Comment by gavinking** [comment]
Assigned to you @FroMage

**Comment by gavinking** [comment]
@fwgreen This bug appears to have fixed itself. Your original code is now working. Closing.

**Comment by gavinking** [comment]
Oh, sorry, my bad. This is an Eclipse-specific bug. Ugh, reopening.

**Comment by fwgreen** [comment]
This problem seems to have resolved itself with the latest plugin (1.3.3).  Closing.

**Comment by gavinking** [comment]
Oh, good, yes, it indeed seems to be fixed.
